usageprocessor/processor.go

```diff
@@ -2,33 +2,55 @@ package usageprocessor
 
 import (
 	"context"
+	"fmt"
 
+	hnyext "github.com/honeycombio/opentelemetry-collector-configs/honeycombextension"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
+var (
+	unset component.ID
+)
+
 type usageProcessor struct {
+	config   *Config
+	recorder hnyext.HoneycombUsageRecorder
 }
 
 func newUsageProcessor() (*usageProcessor, error) {
 	return &usageProcessor{}, nil
 }
 
+func (p *usageProcessor) Start(ctx context.Context, host component.Host) error {
+	if p.config.honeycombExtensionID != unset {
+		ext := host.GetExtensions()[p.config.honeycombExtensionID]
```

Comment:
Thinking about the pattern of having the processor communicate with the extension some more I think we should define the recording interface in the processor and infer that the configured extension implements the processor's interface. This will let us use other implementors if needed, potentially in tests or if we ever want to try to upstream this component.

Reply:
Moved recorder interface to usageprocessor package so we can add tests in [c0491f0](https://github.com/honeycombio/opentelemetry-collector-configs/pull/146/commits/c0491f0722eaef4d5692f3b05e80a48af040169e)